storage/innobase/fil/fil0fil.cc

```diff
@@ -3366,3 +3366,19 @@ fil_space_t *fil_space_t::prev_in_unflushed_spaces()
 }
 
 #endif
+
+/** Returns the size of the space in pages. The tablespace must be cached
+in the memory cache.
+@param[in]      space_id        Tablespace ID
+@return space size, 0 if space not found */
+ulint fil_space_get_size(uint32_t space_id)
```

Comment:
Why is this function being added? How is the return value being used? The return value may already be stale by the time the function returns.

Reply:
This function is used in trx_undo_hdr_get_scn to perform a rough check. If the page exceeds the size of the space, it means that the history of the transaction has been purged.